var/spack/repos/builtin/packages/genomeworks/package.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Genomeworks(CMakePackage, CudaPackage):
+    """SDK for GPU accelerated genome assembly and analysis."""
+
+    homepage = "https://clara-parabricks.github.io/GenomeWorks/"
+    url      = "https://github.com/clara-parabricks/GenomeWorks/archive/v0.5.3.tar.gz"
+    git      = "https://github.com/clara-parabricks/GenomeWorks.git"
+
+    version('0.5.3', tag='v0.5.3', submodules=True)
+    version('0.5.2', tag='v0.5.2', submodules=True)
+    version('0.5.1', tag='v0.5.1', submodules=True)
+    version('0.5.0-rc1', tag='v-.5.0-rc1', submodules=True)
+    version('0.5.0', tag='v0.5.0', submodules=True)
+    version('0.4.4', tag='v0.4.4', submodules=True)
+    version('0.4.3', tag='v0.4.3', submodules=True)
+    version('0.4.0', tag='v0.4.0', submodules=True)
+    version('0.3.0', tag='v0.3.0', submodules=True)
+    version('0.2.0', tag='v0.2.0', submodules=True)
+
+    depends_on('cmake@3.10.2:', type=('build'))
+    depends_on('cuda@11:', type=('build', 'run'))
+    depends_on('python@3.6.7:', type=('build', 'run'))
+
+    phases = ['cmake', 'install']
+    patch('3rdparty.patch')
+
+    def cmake_args(self):
+        args = ["-DCMAKE_BUILD_TYPE=Release"]
```

Comment:
Spack already adds this flag as part of the base package. Does the default work? If not, you can override the variant like other packages do.

Reply:
The default does not work, as I need to add flags for CUDA (line 37 and below). I therefore checked the first 20 packages in var/spack/repos/builtin/. Either the cmake args are not changed, or the whole cmake_args function is re-defined. I also read this way of working in the documentation [here](https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html#adding-flags-to-cmake). Is that correct?

Reply:
@dunatotatos This is not what @adamjstewart meant, I guess. The `cmake_args` method is used here in Spack:

https://github.com/spack/spack/blob/cfd69da8391dbe6acc38ae29274ba3b48bb81ba3/lib/spack/spack/build_systems/cmake.py#L333-L339

The most commonly used options (i.e. which generator to use and the build type) are gathered in the `std_cmake_args` property, therefore there's no need to set `-DCMAKE_BUILD_TYPE=Release`.

**TL;DR** I think Adam meant to just delete line 35, not to use the default `cmake_args` and delete the method completely.

Reply:
Ah, understood! I'll test without the line and update the PR.